File: backend/src/modules/transportation/notifications/notifications.controller.ts
import { Controller, Get, Headers, Post, UseGuards } from '@nestjs/common';
import { HeaderAuthGuard, Validation } from '../../../common/auth/header-auth.guard';
import { CurrentUser } from '../../../common/auth/current-user.decorator';
import { NotificationsService } from './notifications.service';

// Base path /api/Transportation; every action requires CompanyName + UserToken
// via the guard. Notifications belong to the calling user (populated only for
// super admins, who are the approval group).
@Controller('api/Transportation')
@UseGuards(HeaderAuthGuard)
export class NotificationsController {
  constructor(private readonly service: NotificationsService) {}

  @Get('GetNotifications')
  list(
    @CurrentUser() v: Validation,
    @Headers('pageno') pageNo?: string,
    @Headers('noofitems') noOfItems?: string,
  ) {
    return this.service.list(v.userId, Number(pageNo) || 1, Number(noOfItems) || 20);
  }

  @Get('GetUnreadNotificationsCount')
  unread(@CurrentUser() v: Validation) {
    return this.service.unreadCount(v.userId);
  }

  @Post('MarkNotificationRead')
  markRead(@CurrentUser() v: Validation, @Headers('id') id?: string) {
    return this.service.markRead(v.userId, Number(id));
  }

  @Post('MarkAllNotificationsRead')
  markAllRead(@CurrentUser() v: Validation) {
    return this.service.markAllRead(v.userId);
  }
}

File: backend-dotnet/GarasTransport.Api/Modules/Transportation/Notifications/NotificationsController.cs
using GarasTransport.Api.Common;
using Microsoft.AspNetCore.Mvc;

namespace GarasTransport.Api.Modules.Transportation.Notifications;

// Base path /api/Transportation; every action requires CompanyName + UserToken
// via the filter. Notifications belong to the calling user (populated only for
// super admins, who are the approval group).
[Route("api/Transportation")]
[HeaderAuth]
public class NotificationsController : ApiControllerBase
{
    private readonly NotificationsService _service;

    public NotificationsController(NotificationsService service) => _service = service;

    [HttpGet("GetNotifications")]
    public Task<object> List() =>
        _service.List(CurrentUser.UserId, HeaderInt("PageNo", 1), HeaderInt("NoOfItems", 20));

    [HttpGet("GetUnreadNotificationsCount")]
    public Task<object> Unread() => _service.UnreadCount(CurrentUser.UserId);

    [HttpPost("MarkNotificationRead")]
    public Task<object> MarkRead() => _service.MarkRead(CurrentUser.UserId, HeaderInt("Id", 0));

    [HttpPost("MarkAllNotificationsRead")]
    public Task<object> MarkAllRead() => _service.MarkAllRead(CurrentUser.UserId);
}
